indico/modules/events/editing/client/js/editing/timeline/FileManager/util.js

```diff
@@ -48,33 +54,45 @@ async function uploadFile(file, url, onUploadProgress) {
   }
 }
 
+export async function uploadAnExistingFile(url, file) {
+  try {
+    const {data} = await indicoAxios.post(url, snakifyKeys(file));
+    return data;
+  } catch (e) {
+    handleAxiosError(e);
+    return null;
+  }
+}
+
 /**
  * Upload file using axios while triggering all needed actions to update progress.
  *
  * @param {Function} action - the action to be executed after upload is done
  * @param {String} fileTypeId - the id of the File Type
  * @param {Array} acceptedFiles - the "accepted files" array sent by react-dropzone
- * @param {String} uploadURL - the URL to POST the files to
+ * @param {Function} uploadCallback - the function to be called on file upload
  * @param {Function} dispatch - the dispatch function for reducer actions
- * @param {String?} fileId - the ID of the file to modify, if any
+ * @param {String?} replaceFileId - the ID of the file to modify, if any
  * @param {Function} onError - the function to be called on file upload error
+ * @param {Number} fileId - the id of the file, if any, used to cross-reference uploadables
  */
 export function uploadFiles(
   action,
   fileTypeId,
   acceptedFiles,
-  uploadURL,
+  uploadCallback,
   dispatch,
-  fileId = null,
-  onError = null
+  replaceFileId = null,
+  onError = null,
+  fileId = null
```

Comment:
Maybe it's time to pass a singe argument (an object) to this function? Having that many positional arguments is getting really messy!

Reply:
I was starting to think it is getting messy too. I'm thinking about moving the delete - when replacing - to a util function and I should be able to refactor this too.